hummingbot/connector/gateway/clob/gateway_sol_clob.py

```diff
@@ -0,0 +1,1252 @@
+            connector_name: str,
+            chain: str,
+            network: str,
+            wallet_address: str,
+            trading_pairs: List[str] = (),
+            trading_required: bool = True
+    ):
+        """
+        :param connector_name: name of connector on gateway :param chain: refers to a blockchain, e.g. ethereum or
+        avalanche :param network: refers to a network of a particular blockchain e.g. mainnet or kovan :param
+        wallet_address: the address of the eth wallet which has been added on gateway :param trading_pairs: a list of
+        trading pairs :param trading_required: Whether actual trading is needed. Useful for some functionalities or
+            commands like the balance command
+        """
+        self._client_config = None
+        self._connector_name = connector_name
+        self._name = "_".join([connector_name, chain, network])
+        super().__init__(client_config_map)
+        self._chain = chain
+        self._network = network
+        self._trading_pairs = trading_pairs
+        self._tokens = set()
+        [self._tokens.update(set(trading_pair.split("-"))) for trading_pair in trading_pairs]
+        self._wallet_address = wallet_address
+        self._trading_required = trading_required
+        self._ev_loop = asyncio.get_event_loop()
+        self._last_poll_timestamp = 0.0
+        self._last_balance_poll_timestamp = time.time()
+        self._last_est_gas_cost_reported = 0
+        self._allowances = {}
+        self._chain_info = {}
+        self._status_polling_task = None
+        self._get_chain_info_task = None
+        self._auto_approve_task = None
+        self._poll_notifier = None
+        self._native_currency = None
+        self._network_transaction_fee: Optional[TokenAmount] = TokenAmount('SOL', constant.FIVE_THOUSAND_LAMPORTS)
+        self._order_tracker: ClientOrderTracker = ClientOrderTracker(connector=self)
+        self._get_markets_task = None
+        self._markets = None
+        self._auto_create_token_accounts_task = None
+        self._tokens_accounts_created: bool = False
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        global s_logger
+        if s_logger is None:
+            s_logger = logging.getLogger(cls.__name__)
+
+        return cast(HummingbotLogger, s_logger)
+
+    @property
+    def chain(self):
+        return self._chain
+
+    @property
+    def network(self):
+        return self._network
+
+    @property
+    def connector(self):
+        return self._connector_name
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def address(self):
+        return self._wallet_address
+
+    # Added for compatibility
+    @property
+    def connector_name(self):
+        return self._connector_name
+
+    async def all_trading_pairs(self, chain: str, network: str) -> List[str]:
+        # Since the solana tokens trading pairs would be too much, we are returning an empty list here.
```

Comment:
Yes, the system was doing this before, but it was too slow.
Inspecting the problem I've seen it was generating 170,367,756 trading pairs (because of the all possible combinations). For the serum.yml configuration, we have the option to whitelist and/or blacklist markets (or trading pairs), for Solana we haven't implemented it because we didn't need it, but we could check this possibility there as well if returning the empty array cannot be used. Do you have a suggestion on how we could proceed with this?